ttnn/cpp/ttnn/operations/data_movement/pad/device/pad_program_factory.cpp

```diff
@@ -1575,4 +1577,143 @@ operation::ProgramWithCallbacks pad_rm_sharded(
     return {.program = std::move(program), .override_runtime_arguments_callback = override_runtime_args_callback};
 }
 
+operation::ProgramWithCallbacks pad_rm_sharded_width_only(
+    const Tensor& input_tensor,
+    Tensor& output,
+    const tt::tt_metal::LegacyShape& output_tensor_shape,
+    const ttnn::SimpleShape& input_tensor_start,
+    float pad_value) {
+    Program program{};
+
+    TT_ASSERT(
+        output.shard_spec().has_value() and output.shard_spec()->shape[1] == output_tensor_shape[-1],
+        "output must be sharded with shard width equal to output width");
```

Comment:
TT_FATAL and maybe improve error message

Reply:
The output shard spec is set by us up the call stack, so I wanted this to be only a debug assert.